akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/Effect.scala

```diff
@@ -51,14 +52,43 @@ import scala.collection.JavaConverters._
    */
   def unhandled(): Effect[Event, State] = Unhandled.asInstanceOf[Effect[Event, State]]
 
+  /**
+   * Stash the current command. Can be unstashed later with `Effect.thenUnstashAll`
+   * or `EffectFactories.unstashAll`.
+   *
+   * Note that the stashed commands are kept in an in-memory buffer, so in case of a crash they will not be
+   * processed. They will also be discarded if the actor is restarted (or stopped) due to that an exception was
+   * thrown from processing a command or side effect after persisting. The stash buffer is preserved for persist
+   * failures if an `onPersistFailure` backoff supervisor strategy is defined.
+   *
+   * Side effects can be chained with `andThen`.
+   */
+  def stash(): ReplyEffect[Event, State] =
+    Stash.asInstanceOf[Effect[Event, State]].thenNoReply()
+
+  /**
+   * Unstash the commands that were stashed with `EffectFactories.stash`.
+   *
+   * It's allowed to stash messages while unstashing. Those newly added
+   * commands will not be processed by this `unstashAll` effect and have to be unstashed
+   * by another `unstashAll`.
+   *
+   * Side effects can be chained with `andThen`, but note that the side effect is run immediately and not after
+   * processing all unstashed commands.
+   *
+   * @see [[Effect.thenUnstashAll]]
+   */
+  def unstashAll(): Effect[Event, State] =
+    none().andThen(SideEffect.unstashAll())
```

Comment:
Are all these no-arg functions with return types using `()` for Java or is it a style choice?

Reply:
Yes, in java it will always be used with `()`